Build failure when setting custom prefix and libdir

I see this failure on current master when setting custom prefix and libdir:
```
$ make -j8 prefix=/usr libdir=/usr/lib64
[...]
Sysimage built. Summary:
Total ───────  58.547774 seconds 
Base: ───────  25.256778 seconds 43.1387%
Stdlibs: ────  33.288799 seconds 56.8575%
    JULIA usr/lib/julia/sys-o.a
ERROR: Unable to load dependent library libopenblas64_
Message: libopenblas64_: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce type [i.e. "Cannot open shared object: no such file or directory"]
Unable to load "libopenblas64_"
┌ Error: No loaded BLAS libraries were built with ILP64 support
└ @ LinearAlgebra.BLAS ~/Dev/julia/usr/share/julia/stdlib/v1.8/LinearAlgebra/src/blas.jl:154
Quitting.
ERROR: LoadError: SystemError: opening file "/tmp/jl_thjTiZ": Aucun fichier ou dossier de ce type
Stacktrace:
  [1] systemerror(p::String, errno::Int32; extrainfo::Nothing)
    @ Base ./error.jl:174
  [2] #systemerror#68
    @ ./error.jl:173 [inlined]
  [3] systemerror
    @ ./error.jl:173 [inlined]
  [4] open(fname::String; lock::Bool, read::Nothing, write::Nothing, create::Nothing, truncate::Nothing, append::Nothing)
    @ Base ./iostream.jl:293
  [5] open
    @ ./iostream.jl:282 [inlined]
  [6] open(f::Base.var"#354#355"{String}, args::String; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:328
  [7] open
    @ ./io.jl:328 [inlined]
  [8] read
    @ ./io.jl:436 [inlined]
  [9] (::Main.anonymous.var"#1#5"{Set{String}, String})(prec_path::String)
    @ Main.anonymous ~/Dev/julia/contrib/generate_precompile.jl:261
 [10] mktempdir(fn::Main.anonymous.var"#1#5"{Set{String}, String}, parent::String; prefix::String)
    @ Base.Filesystem ./file.jl:750
 [11] mktempdir (repeats 2 times)
    @ ./file.jl:748 [inlined]
 [12] generate_precompile_statements()
    @ Main.anonymous ~/Dev/julia/contrib/generate_precompile.jl:243
 [13] top-level scope
    @ ~/Dev/julia/contrib/generate_precompile.jl:412
 [14] eval(m::Module, e::Any)
    @ Core ./boot.jl:373
 [15] top-level scope
    @ ~/Dev/julia/contrib/generate_precompile.jl:6
in expression starting at /home/milan/Dev/julia/contrib/generate_precompile.jl:3
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [sysimage.mk:88 : /home/milan/Dev/julia/usr/lib/julia/sys-o.a] Erreur 1
make: *** [Makefile:88 : julia-sysimg-release] Erreur 2
```

Cc: @staticfloat as usual :-)

--- a/stdlib/LinearAlgebra/src/LinearAlgebra.jl
+++ b/stdlib/LinearAlgebra/src/LinearAlgebra.jl
@@ -576,8 +576,13 @@ function __init__()
     try
         libblas_path = find_library_path(Base.libblas_name)
         liblapack_path = find_library_path(Base.liblapack_name)
+        # We manually `dlopen()` these libraries here, so that we search with `libjulia-internal`'s
+        # `RPATH` and not `libblastrampoline's`.  Once it's been opened, when LBT tries to open it,
+        # it will find the library already loaded.
+        libblas_path = Libdl.dlpath(Libdl.dlopen(libblas_path))
         BLAS.lbt_forward(libblas_path; clear=true)
         if liblapack_path != libblas_path
+            liblapack_path = Libdl.dlpath(Libdl.dlopen(liblapack_path))
             BLAS.lbt_forward(liblapack_path)
         end
         BLAS.check()

